contrib/clojure-package/src/org/apache/clojure_mxnet/gen/ndarray.clj

```diff
@@ -0,0 +1,2312 @@
+(ns org.apache.clojure-mxnet.ndarray
+  (:refer-clojure :exclude [* - + > >= < <= / cast concat flatten identity load max
+                            min repeat reverse set sort take to-array empty shuffle])
+  (:import (org.apache.mxnet NDArray Shape)))
+
+;; Do not edit - this is auto-generated
+
+;; Licensed to the Apache Software Foundation (ASF) under one or more
+;; contributor license agreements.  See the NOTICE file distributed with
+;; this work for additional information regarding copyright ownership.
+;; The ASF licenses this file to You under the Apache License, Version 2.0
+;; (the "License"); you may not use this file except in compliance with
+;; the License.  You may obtain a copy of the License at
+;;
+;;    http://www.apache.org/licenses/LICENSE-2.0
+;;
+;; Unless required by applicable law or agreed to in writing, software
+;; distributed under the License is distributed on an "AS IS" BASIS,
+;; WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+;; See the License for the specific language governing permissions and
+;; limitations under the License.
+;;
+
+
+
+
```

Comment:
in my understanding, you're listing all the operators below. Can Clojure auto-generate these functions? or can we have a function to allow users call by function name? e.g., `NDArray.call("foo", arg1, arg2, ...,)`.

Reply:
Currently, Clojure generated these by the write file method. I saw @gigasquid uploads the 3000+ lines of code on Symbol and NDArray in the Gen folder.

Reply:
@yzhliu @lanking520 That is correct. We are autogenerating the code in the `gen` directories using the `/dev/generator.clj` code. This code gets invoking using a command line `lein generate-code` and produces these files by inspecting the classes created by Scala with java reflection. Where the scala package creates these classes using macros directly into memory without persisting, the clojure package chose to persist them to files to help the user discover what functions are available, generate api documentation, and to be able to see what new functions were added via a git diff from release to release.

Reply:
Can we generate the source code file during compiling, instead of checking in to the repo?

Reply:
Yes that is definitely a possibility to investigate. There are pros and cons to doing it manually vs compile time. For example, doing it as a manual step gives you a way to control the flow of changes into the project and a git diff. Where, doing it at compile time gives the advantage of not having to worry about a manual step. I'm definitely open to changing this aspect of it, especially if the manual `generate-code` step becomes a burden. I added it to the https://cwiki.apache.org/confluence/display/MXNET/Clojure+Package+Contribution+Needs as feature to further investigate. Thanks for the feedback :)

Reply:
That's a good idea. I added the detail of this to the Clojure Package Contribution Needs - thanks :)

Reply:
@gigasquid I am also concerned that we are merging generated code, we do not want to merge massive amounts of generated code -- this is also a moving target as and when new operators are added or modified. The proposal to only have the interface/documentation also helps us to have a config driven implementation. Do you think you can prioritize this before 1.3?

Reply:
@nswamy Of course. I can work on that next.

Reply:
thanks so much :), you rock!